pkg/provider/certificate_authority.go

```diff
@@ -65,7 +80,30 @@ func (caS CASetupConfig) Status() TLSFilesStatus {
 
 // Create generates and saves a CA using the config
 func (caS CASetupConfig) Create(ctx context.Context) (*FullCertificateAuthority, error) {
-	ca, err := NewCA(ctx, uint16(caS.Difficulty), caS.Concurrency)
+	var (
+		err    error
+		parent *FullCertificateAuthority
+	)
+	if caS.ParentCertPath != "" && caS.ParentKeyPath != "" {
+		parent, err = FullCAConfig{
+			CertPath: caS.ParentCertPath,
+			KeyPath:  caS.ParentKeyPath,
+		}.Load()
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	if parent == nil {
```

Comment:
would it make sense to just intialize it as this to begin with ?

Reply:
`parent` gets assigned on line 88 and it might be nil by line 97